diff -ruN -r glpk-4.50/w32/Makefile_VC glpk-4.50/w32/Makefile_VC
--- glpk-4.50/w32/Makefile_VC	2013-05-24 10:00:00.000000000 +0200
+++ glpk-4.50/w32/Makefile_VC	2013-05-30 13:18:37.000000000 +0200
@@ -4,6 +4,7 @@
 /I. \
 /I..\src \
 /I..\src\amd \
+/I..\src\bflib \
 /I..\src\colamd \
 /I..\src\minisat \
 /I..\src\zlib \
@@ -158,6 +159,7 @@
         lib.exe /out:glpk.lib \
                 ..\src\*.obj \
                 ..\src\amd\*.obj \
+                ..\src\bflib\*.obj \
                 ..\src\colamd\*.obj \
                 ..\src\minisat\*.obj \
                 ..\src\zlib\*.obj
diff -ruN -r glpk-4.50/w32/Makefile_VC_DLL glpk-4.50/w32/Makefile_VC_DLL
--- glpk-4.50/w32/Makefile_VC_DLL	2013-05-24 10:00:00.000000000 +0200
+++ glpk-4.50/w32/Makefile_VC_DLL	2013-05-30 13:18:28.000000000 +0200
@@ -4,6 +4,7 @@
 /I. \
 /I..\src \
 /I..\src\amd \
+/I..\src\bflib \
 /I..\src\colamd \
 /I..\src\minisat \
 /I..\src\zlib \
@@ -158,6 +159,7 @@
         cl.exe $(CFLAGS) /LD /Feglpk_4_50.dll \
                 ..\src\*.obj \
                 ..\src\amd\*.obj \
+                ..\src\bflib\*.obj \
                 ..\src\colamd\*.obj \
                 ..\src\minisat\*.obj \
                 ..\src\zlib\*.obj \
diff -ruN -r glpk-4.50/w64/makefile_VC glpk-4.50/w64/makefile_VC
--- glpk-4.50/w64/makefile_VC	2013-05-24 10:00:00.000000000 +0200
+++ glpk-4.50/w64/makefile_VC	2013-05-30 13:16:49.000000000 +0200
@@ -4,6 +4,7 @@
 /I. \
 /I..\src \
 /I..\src\amd \
+/I..\src\bflib \
 /I..\src\colamd \
 /I..\src\minisat \
 /I..\src\zlib \
@@ -158,6 +159,7 @@
         lib.exe /out:glpk.lib \
                 ..\src\*.obj \
                 ..\src\amd\*.obj \
+                ..\src\bflib\*.obj \
                 ..\src\colamd\*.obj \
                 ..\src\minisat\*.obj \
                 ..\src\zlib\*.obj
diff -ruN -r glpk-4.50/w64/makefile_VC_DLL glpk-4.50/w64/makefile_VC_DLL
--- glpk-4.50/w64/makefile_VC_DLL	2013-05-24 10:00:00.000000000 +0200
+++ glpk-4.50/w64/makefile_VC_DLL	2013-05-30 13:17:12.000000000 +0200
@@ -4,6 +4,7 @@
 /I. \
 /I..\src \
 /I..\src\amd \
+/I..\src\bflib \
 /I..\src\colamd \
 /I..\src\minisat \
 /I..\src\zlib \
@@ -158,6 +159,7 @@
         cl.exe $(CFLAGS) /LD /Feglpk_4_50.dll \
                 ..\src\*.obj \
                 ..\src\amd\*.obj \
+                ..\src\bflib\*.obj \
                 ..\src\colamd\*.obj \
                 ..\src\minisat\*.obj \
                 ..\src\zlib\*.obj \
